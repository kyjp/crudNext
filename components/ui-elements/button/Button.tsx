import React, { ComponentPropsWithoutRef, FC } from 'react'

type ButtonType = ComponentPropsWithoutRef<'button'>

const Button: FC<ButtonType> = (props) => {
    const { className, disabled, onClick, type, children } = props
    return (
        <button
            className={`text-white bg-blue-700 hover:bg-blue-800 focus:outline-none focus:ring-4 focus:ring-blue-300 font-medium rounded-full text-sm px-5 py-2.5 text-center me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800${className && ` ${className}`}`}
            onClick={onClick}
            type={type}
            disabled={disabled}
        >
            {children}
        </button>
    )
}

export default Button