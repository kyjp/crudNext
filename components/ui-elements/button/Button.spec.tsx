import { render, screen, waitFor } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import '@testing-library/jest-dom'
import Button from "./Button"

describe('Test group of Button component', () => {
    const text = "This is Button"
    // snapshot test
    test("Snapshot Test > renders Button component", () => {
        const { asFragment } = render(
            <Button>
                {text}
            </Button>
        )
        expect(asFragment()).toMatchSnapshot()
    })
    
    test("renders Button component", () => {
        render(
            <Button>
                {text}
            </Button>
        )
        expect(screen.getByRole('button')).toBeInTheDocument()   
    })
    
    test("when clicked, 'mockFn' is called once", async () => {
        const mockFn = vi.fn()
        render(
            <Button          
                onClick={() => {mockFn()}}
            >
                {text}
            </Button>
        )
        const ButtonComponent = screen.getByRole('button')
        expect(ButtonComponent).toBeInTheDocument()
        await waitFor(() => userEvent.click(ButtonComponent))
        expect(mockFn).toBeCalled()
        expect(mockFn).toBeCalledTimes(1)
    })
})