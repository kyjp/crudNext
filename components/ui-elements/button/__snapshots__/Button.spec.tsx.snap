// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test group of Button component > Snapshot Test > renders Button component 1`] = `
<DocumentFragment>
  <button
    class="text-white bg-blue-700 hover:bg-blue-800 focus:outline-none focus:ring-4 focus:ring-blue-300 font-medium rounded-full text-sm px-5 py-2.5 text-center"
  >
    This is Button
  </button>
</DocumentFragment>
`;
